torchdata/datapipes/iter/util/webdataset.py

```diff
@@ -0,0 +1,66 @@
+import re
+from typing import Dict, Iterator, List, Union
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+def pathsplit(p):
+    """Split a path into the basename and the extensions."""
+    if "." not in p:
+        return p, ""
+    prefix, suffix = re.search(r"^(.*?)(\.[^/]*)$", p).groups()
+    return prefix, suffix
+
+
+@functional_datapipe("webdataset")
+class WebDatasetIterDataPipe(IterDataPipe[Dict]):
+    r"""
+    Iterable DataPipe that accepts stream of (path, data) tuples and aggregates them into
+    a single dictionary based on the basenames of the paths (WebDataset file convention).
+
```

Comment:
Please mention that it is aggregates only sequential data, meaning if same key is going to be separated by different one, it will yield 3 data pieces.

Reply:
I have updated the comment and added the commit to this PR.